### ISSUE TITLE
Dex fails on startup due to a missing secret in the argo-cd-cli configuration

### ISSUE THREAD

**Issue by MaSer713** [argoproj/argo-cd#3519]
**Describe the bug**

argocd-util rundex fails on startup with invalid config: Secret or SecretEnv field is required for client "argo-cd-cli"

**To Reproduce**

Here is the dex configuration clause we use
```
dex.config: |
    connectors:
    - type: google
      id: google
      name: Google
      config:

        clientID: $dex.clientid
        clientSecret: $dex.clientsecret

        redirectUri: .../api/dex/callback

        hostedDomains:
         - ...

        serviceAccountFilePath: /etc/argocd/dex/googleAuth.json
        adminEmail: $dex.adminemail
```

Dex is functioning correctly if we manually amend the resulting dex.yaml configuration and remove the argo-cd-cli client from it or give it any secret value, but the cli SSO is not functional in this case.

**Expected behavior**

Dex does not fail on startup and both SSO methods (web and cli) are working correctly

**Version**

argocd version
```shell
argocd: v1.5.2+c2c19f4
  BuildDate: 2020-04-15T16:45:04Z
  GitCommit: c2c19f42ad78ed7a6fb70e86aed117be484feb50
  GitTreeState: clean
  GoVersion: go1.14
  Compiler: gc
  Platform: windows/amd64
argocd-server: v1.5.2+c2c19f4
  BuildDate: 2020-04-15T16:43:12Z
  GitCommit: c2c19f42ad78ed7a6fb70e86aed117be484feb50
  GitTreeState: clean
  GoVersion: go1.14
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: {Version:unknown GitCommit:$Format:%H$ BuildDate:1970-01-01T00:00:00Z GoOs:linux GoArch:amd64}
  Helm Version: version.BuildInfo{Version:"v3.1.1", GitCommit:"afe70585407b420d0097d07b21c47dc511525ac8", GitTreeState:"clean", GoVersion:"go1.13.8"}
  Kubectl Version: v1.14.0
```

argocd-util version
```
argocd-util: v1.5.2+c2c19f4
  BuildDate: 2020-04-15T16:43:36Z
  GitCommit: c2c19f42ad78ed7a6fb70e86aed117be484feb50
  GitTreeState: clean
  GoVersion: go1.14
  Compiler: gc
  Platform: linux/amd64
```

dex version (the image is 2.23.0)
```
dex Version: 0a85a97ba9d85fa0de8542a68f1edbc53ed77c25
Go Version: go1.13.10
Go OS/ARCH: linux amd64
```

**Logs**

This is what argocd-util rundex results in
```
INFO[0000] Starting configmap/secret informers
INFO[0000] Configmap/secret informer synced
INFO[0000] 0xc000071a40 subscribed to settings updates
time="2020-04-30T08:41:42Z" level=info msg="config issuer: .../api/dex"
time="2020-04-30T08:41:42Z" level=info msg="config storage: memory"
time="2020-04-30T08:41:42Z" level=info msg="config static client: Argo CD"
invalid config: Secret or SecretEnv field is required for client "argo-cd-cli"
```

**Comment by MaSer713** [comment]
Updated to the current master, the bug seems to be gone with the dex image version 2.22.0

**Comment by santinoncs** [comment]
This is not fixed in 2.23.0 
It is in 2.24.0
